### PR TITLE
16th task 2408

### PR DESCRIPTION
**Pull request by Clasher370**

### CODE REVIEW

File: app/controllers/oauths_controller.rb [hunk 1]
```diff
@@ -8,15 +8,15 @@ def oauth
   def callback
     provider = auth_params[:provider]
     if @user = login_from(provider)
-      redirect_to root_path, :notice => "Logged in from #{provider.titleize}!"
+      redirect_to root_path, :notice => t('.login', provider: provider.titleize)
```
[the remark this answers is not in the document]
Comment (X1ting) [reply app/controllers/oauths_controller.rb]: agree with it. 
`redirect_to root_path, notice: t('.login' ....`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -6,14 +6,15 @@ def new
   end
 
   def edit
+    @user = current_user
     @decks = current_user.decks
```
Comment (X1ting) [inline app/controllers/users_controller.rb]: why not `current_user` in view?

File: db/migrate/20170825054348_add_locale_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddLocaleToUsers < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :locale, :string
```
Comment (X1ting) [inline db/migrate/20170825054348_add_locale_to_users.rb]: you add this field, and made it required ( validataions) but not adding default value. It mean, what after this migration, you will have invalid records in db (without locale)

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Clasher370) [reply app/controllers/users_controller.rb]: Use @user instance in _form partial for new and edit action

Comment (X1ting) [reply app/controllers/users_controller.rb]: current_user unavaliable in _form partial for new end edit action?

**Pull request merged**